### PR TITLE
feat: Add courses section

### PR DESCRIPTION
**Pull request by TarekMGamal**

### CODE REVIEW

File: src/components/coursesSection/CoursesSection.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import React from "react";
+import Courses from "./courses/Courses";
+import coursesSectionStyle from "./CoursesSection.module.css";
+
+function coursesSection() {
```
Comment (AhmedMorgan) [inline src/components/coursesSection/CoursesSection.js]: - the component name should start with an uppercase letter.
- title, description, and course list should be passed to this component as props.

File: src/components/coursesSection/course/Course.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import courseStyle from "./Course.module.css";
+
+function Course(props) {
+  return (
+    <div className={courseStyle.course}>
+      <img className={courseStyle.courseImage} src={props.courseData.image} alt="course"/>
+      <h3>{props.courseData.title}</h3>
+      <div>{props.courseData.author}</div>
+      <div>
+        {props.courseData.rating + "/5 (" + props.courseData.people + ")"}
```
Comment (AhmedMorgan) [inline src/components/coursesSection/course/Course.js]: the rating should be represented with stars also and it should be dynamic. You can use a third party lib or create your customized component.

File: src/components/coursesSection/course/Course.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import courseStyle from "./Course.module.css";
+
+function Course(props) {
```
Comment (AhmedMorgan) [inline src/components/coursesSection/course/Course.js]: it is better to receive the props in this format "function Course({ courseData })"

File: src/components/coursesSection/courses/Courses.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
+import React from "react";
+import Course from "../course/Course";
+import coursesStyle from "./Courses.module.css";
+
+function Courses() {
```
Comment (AhmedMorgan) [inline src/components/coursesSection/courses/Courses.js]: this component should accept the courses as a prop.

File: src/components/coursesSection/course/Course.module.css [hunk 5]
```diff
@@ -0,0 +1,11 @@
+.course {
```
Comment (AhmedMorgan) [inline src/components/coursesSection/course/Course.module.css:1]: it is better here to give the course and its image max-width for better rendering on the smaller screens.

File: src/components/coursesSection/CoursesSection.js (hunk 1 above)
Comment (TarekMGamal) [reply src/components/coursesSection/CoursesSection.js]: * I assumed that the component name is the name of the function, if it is not the case, please let me know
* I placed the title, description, and courses list in the index.js file and passed them as props to their required position

File: src/components/coursesSection/course/Course.js (hunk 3 above)
Comment (TarekMGamal) [reply src/components/coursesSection/course/Course.js]: Done

File: src/components/coursesSection/courses/Courses.js (hunk 4 above)
Comment (TarekMGamal) [reply src/components/coursesSection/courses/Courses.js]: Done

File: src/components/coursesSection/course/Course.js (hunk 2 above)
Comment (TarekMGamal) [reply src/components/coursesSection/course/Course.js]: Done

File: src/components/coursesSection/course/Course.module.css (hunk 5 above)
Comment (TarekMGamal) [reply src/components/coursesSection/course/Course.module.css:1]: Done

File: src/components/coursesSection/CoursesSection.js (hunk 1 above)
Comment (AhmedMorgan) [reply src/components/coursesSection/CoursesSection.js]: according to title, description, and courses I mean you pass them from app.js to this component not from index.js but it is ok for now. please fix it in the next task.

**Pull request merged**